### PR TITLE
Packages: Further clarify the npm publishing process

### PR DESCRIPTION
**Pull request by gziolo**
## Description

This PR tries to clarify the documentation for npm publishing. In particular, it's targeting the upcoming minor WordPress release.

The best way to see changes is to visit:
https://github.com/WordPress/gutenberg/pull/30733/files?short_path=17d1ffe#diff-17d1ffea91f685c145076051d0b46384b2cf600635cc9429d877e168914aca1b
and
https://github.com/WordPress/gutenberg/pull/30733/files?short_path=1206c4b#diff-1206c4b8d1f7e37d1a4079860d4b24c3178398a059b3bd4c0df5c40fabaab45a

### CODE REVIEW

File: packages/README.md [hunk 1]
```diff
@@ -176,7 +176,7 @@ If you are publishing new versions of packages, note that there are versioning r
 
 ## Releasing Packages
 
-Lerna automatically releases all outdated packages. To check which packages are outdated and will be released, type `npm run publish:check`.
+Lerna automatically releases all outdated packages. To check which packages are outdated and will be released, type `npm run publish:check` from the branch that keeps the code for a given release type.
```
Comment (ntsekouras) [inline packages/README.md]: Maybe add the example here as well? 🤔 

>`wp/X.Y` (example `wp/5.7`) release branch

**Review by ntsekouras — APPROVED** [review]
Always great to enhance docs 💯 - thanks!!

File: packages/README.md (hunk 1 above)
Comment (gziolo) [inline packages/README.md]:
```suggestion
Lerna automatically releases all outdated packages. To check which packages are outdated and will be released, type `npm run publish:check` from the branch that keeps the code for a given `wp/X.Y` release type (example `wp/5.7`).
```

Comment (gziolo) [reply packages/README.md]: Added in https://github.com/WordPress/gutenberg/pull/30733/commits/c0228b5afa5956ee684821f211ca0f9cc26d4721.

**Pull request merged**